js/plot/util/fill.js

```diff
@@ -121,6 +123,54 @@ module.exports = function(pool, opts) {
         });
 
       fills.exit().remove();
+
+      // Add midnight markers
+      if (opts.isDaily) {
+        selection.selectAll('rect.d3-fill-midnight')
+        .data(_.filter(currentData, {startsAtMidnight: true}))
```

Comment:
For safety & consistency, can you add the [key function](https://github.com/mbostock/d3/wiki/Selections#data) like I did for the timechange markers in #228 (and like is above in line 61)?